src/App.js

```diff
@@ -1,16 +1,44 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
+
+const chatLogData = chatMessages;
 
 const App = () => {
+  const [updatedChatLogData, setChangeLikeState] = useState(chatLogData)
+
+  const onChangeLike = (id) => {
+    setChangeLikeState(() => updatedChatLogData.map(chat => {
+      if (chat.id === id) {
+        return { ...chat, liked: !chat.liked };
+      }
+      else {
```

Comment:
We could remove the else & unindent the default return without changing the outcome.